Tasks/BuildctlV0/src/buildctl.ts

```diff
@@ -0,0 +1,91 @@
+"use strict";
+
+import tl = require('azure-pipelines-task-lib/task');
+import path = require('path');
+import * as toolLib from 'azure-pipelines-tool-lib/tool';
+import utils = require("./utils");
+
+
+tl.setResourcePath(path.join(__dirname, '..', 'task.json'));
+
+async function configureBuildctl() {
+    var stableBuildKitVersion = await utils.getBuildctlVersion();
+    var buildctlPath = await utils.downloadBuildctl(stableBuildKitVersion);
+
+    // prepend the tools path. instructs the agent to prepend for future tasks
+    if (!process.env['PATH'].startsWith(path.dirname(buildctlPath))) {
+        toolLib.prependPath(path.dirname(buildctlPath));
+    }
+}
+
+async function verifyBuildctl() {
+    tl.debug(tl.loc("VerifyBuildctlInstallation"));
+    var buildctlToolPath = tl.which("buildctl", true);
+    var buildctlTool = tl.tool(buildctlToolPath);
+    buildctlTool.arg("--help");
+    buildctlTool.exec();
+}
+    
+async function buildContainer() {
+    if(process.env["RUNNING_AS_K8_POOLPROVIDER"] == "1") {
```

Comment:
Change env variable to RUNNING_ON. Possible values right now will be KUBERNETES. In future we'll differentiate between VMs, ACI and so on...

Reply:
Changed to RUNNING_ON